xonsh/ptk_shell/completer.py

```diff
@@ -99,6 +101,35 @@ def get_completions(self, document, complete_event):
                 disp = comp[pre:].strip("'\"")
                 yield Completion(comp, -plen, display=disp)
 
+    def __find_begidx(self, line, endidx):
+        if len(line[0:endidx]) == 0:
+            return 0
+
+        wordsNonPosix = []
+        try:
+            wordsNonPosix = split(
+                line[0:endidx] + "_", posix=False
+            )  # ensure last word is empty when ends with space
+            wordsNonPosix[-1] = wordsNonPosix[-1][:-1]
```

Comment:
This seems a little bit circuitous -- what's the end goal here?
Make sure that the last element in `words_non_posix` is an empty string if `line[0:endidx] == 0`?
It's better to be explicit if that's what you're after -- something like

```
if line[0:endidx].endswith(" "):
    words_non_posix.append("")
```

Reply:
Yeah, been a while that i have written this and i think it was just because i initially had some issues with split and also needed an additional (empty) argument passed to carapace for the completion to work correctly.
Just had another look at it and while it shouldn't be needed anymore here it does make the loop below a bit easier.
Ah, might work to move the check downward, let's see..